sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java

```diff
@@ -99,6 +106,90 @@ public PageBlobClient getPageBlobClient() {
+     * overwrites any existing metadata on the blob. Partial updates are not supported with this method; the content of
+     * the existing blob is overwritten with the new content. To perform a partial update of a block blob's, use {@link
+     * BlockBlobAsyncClient#stageBlock(String, Flux, long) stageBlock} and
+     * {@link BlockBlobAsyncClient#commitBlockList(List)}, which this method uses internally. For more information,
+     * see the <a href="https://docs.microsoft.com/rest/api/storageservices/put-block">Azure
+     * Docs for Put Block</a> and the <a href="https://docs.microsoft.com/rest/api/storageservices/put-block-list">Azure
+     * Docs for Put Block List</a>.
+     * <p>
+     * The data passed need not support multiple subscriptions/be replayable as is required in other upload methods when
+     * retries are enabled, and the length of the data need not be known in advance. Therefore, this method should
+     * support uploading any arbitrary data source, including network streams. This behavior is possible because this
+     * method will perform some internal buffering as configured by the blockSize and numBuffers parameters, so while
+     * this method may offer additional convenience, it will not be as performant as other options, which should be
+     * preferred when possible.
+     * <p>
+     * Typically, the greater the number of buffers used, the greater the possible parallelism when transferring the
+     * data. Larger buffers means we will have to stage fewer blocks and therefore require fewer IO operations. The
+     * trade-offs between these values are context-dependent, so some experimentation may be required to optimize inputs
+     * for a given scenario.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobClient.uploadWithResponse#InputStream-ParallelTransferOptions-BlobHTTPHeaders-Map-AccessTier-BlobAccessConditions-Duration}
+     *
+     * @param data The data to write to the blob.
+     * @param parallelTransferOptions {@link ParallelTransferOptions} used to configure buffered uploading.
+     * @param headers {@link BlobHTTPHeaders}
+     * @param metadata Metadata to associate with the blob.
+     * @param tier {@link AccessTier} for the destination blob.
+     * @param accessConditions {@link BlobAccessConditions}
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
+     * @return A reactive response containing the information of the uploaded block blob.
+     */
+    public Response<BlockBlobItem> uploadWithResponse(InputStream data, ParallelTransferOptions parallelTransferOptions,
+        BlobHTTPHeaders headers, Map<String, String> metadata, AccessTier tier, BlobAccessConditions accessConditions,
+        Duration timeout) {
+
+        Flux<ByteBuffer> fbb = Utility.convertStreamToByteBuffer(data, BlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE);
```

Comment:
Instead of calling into the `Utility` conversion method we should just chunk the `InputStream` into `ByteBuffer`s the size of `ParallelTransferOptions.getBlockSize`. It'll need to be a conversion that runs until the `InputStream` is depleted and indicates end of file/stream.